rust/sealevel/programs/interchain-accounts/src/accounts.rs

```diff
@@ -0,0 +1,86 @@
+//! InterchainAccount accounts.
+use std::collections::HashMap;
+
+use access_control::AccessControl;
+use account_utils::{AccountData, SizedData};
+use borsh::{BorshDeserialize, BorshSerialize};
+use hyperlane_core::H256;
+use hyperlane_sealevel_connection_client::{
+    router::{HyperlaneRouter, RemoteRouterConfig},
+    HyperlaneConnectionClient,
+};
+use hyperlane_sealevel_igp::accounts::InterchainGasPaymasterType;
+
+use solana_program::{program_error::ProgramError, pubkey::Pubkey};
+
+/// The storage account.
+pub type InterchainAccountStorageAccount = AccountData<InterchainAccountStorage>;
+
+/// The storage account's data.
+#[derive(BorshSerialize, BorshDeserialize, Debug, Default)]
+pub struct InterchainAccountStorage {
+    /// The local domain.
+    pub local_domain: u32,
+    /// The mailbox.
+    pub mailbox: Pubkey,
+    /// The ISM.
+    pub ism: Option<Pubkey>,
+    /// The IGP.
+    pub igp: Option<(Pubkey, InterchainGasPaymasterType)>,
+    /// The owner.
+    pub owner: Option<Pubkey>,
+    /// Keyed by domain, the router for the remote domain.
```

Comment:
fwiw us keeping routers in the hashmap here means that we have some limits on how large the routers can get. I think it's something like 50 where we max out. Might be worth us considering a more scalable approach in this new program, or we can always make the changes later. The scalable approach would be to store each router in its own account

Reply:
we actually can allow the caller to specify their own destination rotuer